### PR TITLE
Add scientific keyword to savename

### PR DESCRIPTION
**Pull request by sebastianpech**
Implements #63

### CODE REVIEW

File: test/naming_tests.jl [hunk 1]
```diff
@@ -80,3 +80,22 @@ x = A(5, NamedTuple())
...
+c = 1
+d = "test"
+di = @dict a b c d
+
+@test savename(di,scientific=6) == "a=1.2345e-7_b=1.0_c=1_d=test"
+@test savename(di,scientific=5) == "a=1.2345e-7_b=1.0_c=1_d=test"
+@test savename(di,scientific=4) == "a=1.234e-7_b=1.0_c=1_d=test"
+@test savename(di,scientific=3) == "a=1.23e-7_b=1.0_c=1_d=test"
+@test savename(di,scientific=2) == "a=1.2e-7_b=1.0_c=1_d=test"
+@test savename(di,scientific=1) == "a=1.0e-7_b=1.0_c=1_d=test"
```
Comment (Datseris) [inline test/naming_tests.jl]: In the case of `scientific = 1`, isn't the second digit always `0` and thus can be skipped?

Comment (sebastianpech) [reply test/naming_tests.jl]: Yes it is. Do you think it's better to fallback to the rounding behavior of the default rounding method? So also adding the check with Integer rounding.

Comment (sebastianpech) [reply test/naming_tests.jl]: In favor of consistency it's better to remove it

Comment (Datseris) [reply test/naming_tests.jl]: > Yes it is. Do you think it's better to fallback to the rounding behavior of the default rounding method?

Yes I think so.

Comment (Datseris) [reply test/naming_tests.jl]: (to be clear I think that `1.0e-7` should be written as `1e-7`. But if this is too hard to do programatically I can live with it not happening.

Comment (sebastianpech) [reply test/naming_tests.jl]: Ah, now I get it. That shouldn't be too difficult.

**Pull request merged**